pkg/endpoint/endpoint.go

```diff
@@ -516,7 +519,7 @@ func createEndpoint(owner regeneration.Owner, policyGetter policyRepoGetter, nam
 	ep.aliveCancel = cancel
 	ep.aliveCtx = ctx
 
-	ep.realizedPolicy = ep.desiredPolicy
+	ep.realizedPolicy = ep.desiredPolicy.DeepCopy()
```

Comment:
While I understand this is probably technically the most correct thing to do, a deep-copy is quite expensive, especially in the hot path. Could we explore a cheaper alternative to exceeding 100%? Maybe during the metrics-update, we check whether the current size exceeds the max, and then truncate?

Reply:
I was a bit surprised to find this, but from what I can tell realizedPolicy and desiredPolicy always point at the same thing.  As a result, realizedPolicy is basically functioning the same as desiredPolicy - perhaps we could just get rid of realized policy?

But I'm missing some context here so I'd like to see what Joe or Jarno think.